x/lockup/keeper/admin_keeper.go

```diff
@@ -0,0 +1,32 @@
+package keeper
+
+import (
+	"github.com/c-osmosis/osmosis/x/lockup/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// RageQuit unlock previous lockID and create a new lock with newCoins with same duration and endtime
+func (ak AdminKeeper) RageQuit(ctx sdk.Context, lockID uint64, newCoins sdk.Coins) error {
+	lock, err := ak.GetLockByID(ctx, lockID)
+	if err != nil {
+		return err
+	}
+
+	// send original coins back to owner
+	if err := ak.bk.SendCoinsFromModuleToAccount(ctx, types.ModuleName, lock.Owner, lock.Coins); err != nil {
+		return err
+	}
+
+	// lock newCoins into module account
+	if ak.bk.SendCoinsFromAccountToModule(ctx, lock.Owner, types.ModuleName, newCoins); err != nil {
```

Comment:
It looks like this line doesn't check the error properly?

Reply:
Good catch